src/CsvWriter/CsvWriterFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\AzureStorageTableExtractor\CsvWriter;
+
+use Keboola\AzureStorageTableExtractor\Configuration\Config;
+use Keboola\AzureStorageTableExtractor\Configuration\ConfigDefinition;
+use Keboola\AzureStorageTableExtractor\Exception\UnexpectedValueException;
+use Keboola\AzureStorageTableExtractor\IncrementalFetchingHelper;
+
+class CsvWriterFactory
+{
+    private string $dataDir;
+
+    private Config $config;
+
+    private IncrementalFetchingHelper $incFetchingHelper;
+
+    public function __construct(string $dataDir, Config $config, IncrementalFetchingHelper $incFetchingHelper)
+    {
+        $this->dataDir = $dataDir;
+        $this->config = $config;
+        $this->incFetchingHelper = $incFetchingHelper;
+    }
+
+    public function create(): ICsvWriter
+    {
+        switch ($this->config->getMode()) {
```

Comment:
Je to zaujimava databaza. Uklada data ako tabulky, ale nema definovane stlpce .... teda dostanem napr `{a:b, c:d}` ...  a v dalsej polozke dostanem `{a:c, e:x}` ... dokonca hodnota moze byt raz int a inokedy napr string, ... preto tu podporujem RAW mode a MAPPING mode ... teda spravam sa k riadku ako ku JSONu